File: frontend/src/api/holiday.ts
import request from './request'

export interface Holiday {
  id?: number
  name: string
  type: 'FIXED_SOLAR' | 'FIXED_LUNAR' | 'DYNAMIC_RULE'
  fixedDate?: string
  lunarMonth?: number
  lunarDay?: number
  dynamicRule?: string
  dynamicMonth?: number
  dynamicWeekOrdinal?: number
  dynamicDayOfWeek?: number
  enabled: boolean
  notifyEmail?: string
  createdAt?: string
  updatedAt?: string
}

// 获取节日列表
export function getHolidays() {
  return request.get('/holidays')
}

// 创建节日
export function createHoliday(data: Holiday) {
  return request.post('/holidays', data)
}

// 更新节日
export function updateHoliday(id: number, data: Holiday) {
  return request.put(`/holidays/${id}`, data)
}

// 删除节日
export function deleteHoliday(id: number) {
  return request.delete(`/holidays/${id}`)
}

File: backend/src/main/java/com/notifyx/entity/Holiday.java
package com.notifyx.entity;

import com.notifyx.enums.HolidayType;
import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "holidays")
public class Holiday {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(length = 50, unique = true)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(length = 20)
    private HolidayType type;

    @Column(length = 10)
    private String fixedDate;

    private Integer lunarMonth;

    private Integer lunarDay;

    @Column(length = 100)
    private String dynamicRule;

    private Integer dynamicMonth;

    private Integer dynamicWeekOrdinal;

    private Integer dynamicDayOfWeek;

    @Column(columnDefinition = "boolean default true")
    private Boolean enabled = true;

    @Column(length = 100)
    private String notifyEmail;

    @CreationTimestamp
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;
}
